src/Http/Response.php

```diff
@@ -53,6 +54,13 @@ class Response
      */
     protected array $decodedJson;
 
+    /**
+     * The decoded JSON response object.
+     *
+     * @var object<object-key, mixed>
+     */
+    protected object $decodedJsonObject;
```

Comment:
This variable shouldn't be typehinted as an `object`. The json decoded result may not necessarily be an object. For example, if the response is `[]`, this will decode as an array, and throw an error in the `object()` method. This was one of the original issues with the return typehint of the `object()` method being `object`; it would throw an error when the API returned a properly indexed array.